### ISSUE TITLE
[native-image] adding --language:js generates js interpreter instead of application 

### ISSUE THREAD

**Issue by lburgazzoli** [oracle/graal#1064]
I'm having an issue with rc13 and using truffle js in a native compilation as the native-image-maven-plugin reports that I'm running:

```
native-image -cp ... -H:Class=org.apache.camel.graalvm.Application -H:Name=camel-routes-loader-graalvm --verbose --language:js -H:+JNI -H:ReflectionConfigurationFiles=../src/graalvm/reflectionconfig.json -H:IncludeResources=META-INF/.* -H:Features=org.apache.camel.graalvm.support.CamelFeature -DCamelSimpleLRUCacheFactory=true
```

but then the build request changes the values of the -H:Class and -H:Name flags to:

```
-H:Name=js
-H:Class=com.oracle.truffle.js.shell.JSLauncher
```

so as result I have the js interpreter built instead of my application

**Comment by olpaw** [comment]
That's by design. You have to provide **your** Class and Name arguments after all the `--language:...` and `--tool:...` options.

**Comment by lburgazzoli** [comment]
ok so the order matters

**Comment by olpaw** [comment]
Btw, you do not really need the `-H:Class=` and `-H:Name=` prefixes here. The following should do what what you need:
```
native-image -cp ... --verbose --language:js -H:+JNI -H:ReflectionConfigurationFiles=../src/graalvm/reflectionconfig.json -H:IncludeResources=META-INF/.* -H:Features=org.apache.camel.graalvm.support.CamelFeature -DCamelSimpleLRUCacheFactory=true
org.apache.camel.graalvm.Application camel-routes-loader-graalvm
```

**Comment by olpaw** [comment]
... and `-H:+JNI` is also redundant nowadays (it's enabled by default)

**Comment by lburgazzoli** [comment]
I do not set the class name, it is the native-image-maven-plugin that does so, what I have is

```
<plugin>
    <groupId>com.oracle.substratevm</groupId>
    <artifactId>native-image-maven-plugin</artifactId>
    <version>${graalvm.version}</version>
    <executions>
        <execution>
            <goals>
                <goal>native-image</goal>
            </goals>
            <phase>package</phase>
            <configuration>
                <mainClass>${main.class}</mainClass>
                <buildArgs>
                    --verbose --language:js -H:+JNI -H:IncludeResources=META-INF/services/.* -H:Features=org.apache.camel.graalvm.support.CamelFeature -DCamelSimpleLRUCacheFactory=true
                </buildArgs>
            </configuration>
        </execution>
    </executions>
</plugin>
~~~

**Comment by olpaw** [comment]
That's ok then. I just thought you also use it when you issue the native-image command yourself

**Comment by olpaw** [comment]
Btw, the native-image-maven-plugin should actually ensure that `-H:Class=` and `-H:Name=` always come last. Is this not the case in your example?

**Comment by lburgazzoli** [comment]
correct, it is not the case in my example

**Comment by olpaw** [comment]
oh. this needs fixing then. thanks for reporting.

**Comment by olpaw** [comment]
Fixed for rc14 and on master (see 9088ea00f618a6864f95b5bdcba6ad0f8c15f454). I have also added `<imageName>` in case you want the image to be named other than the default. Once rc14 is released you can remove `-H:Class=${main.class} -H:Name=${project.artifactId}` from your `<buildArgs>` and just add `<imageName>${project.artifactId}</imageName>` for your custom image name. The `<mainClass>` will now be correctly inferred from your `<maven-jar-plugin>/<archive>/<manifest>/<mainClass>`. No need to specify <mainClass> explicitly anymore:
```
@@ -162,10 +143,10 @@
                                 </goals>
                                 <phase>package</phase>
                                 <configuration>
-                                    <mainClass>${main.class}</mainClass>
                                     <buildArgs>
-                                        --verbose --language:js -H:IncludeResources=META-INF/services/.* -H:Features=org.apache.camel.graalvm.support.CamelFeature -DCamelSimpleLRUCacheFactory=true -H:Class=${main.class} -H:Name=${project.artifactId}
+                                        --verbose --language:js -H:IncludeResources=META-INF/services/.* -H:Features=org.apache.camel.graalvm.support.CamelFeature -DCamelSimpleLRUCacheFactory=true
                                     </buildArgs>
+                                    <imageName>${project.artifactId}</imageName>
                                 </configuration>
                             </execution>
                         </executions>
```
